### ISSUE TITLE
In `Select` Component, when using `children` as a function, nothing renders.

### ISSUE THREAD

**Issue by meesvandongen** [adobe/react-spectrum#4562]
# 🐛 Bug Report 

Nothing shows up on the screen when using the `children` callback of `react-aria-component`'s `Select`. 

```jsx
<Select>
  {(values) => (
    <>
      <Button>
        <SelectValue />
      </Button>
    </>
  )}
</Select>
```

Note that when not using the callback, and directly returning JSX, it does work.

```jsx
<Select>
  <>
    <Button>
      <SelectValue />
    </Button>
  </>
</Select>
```

Maybe I am missing something. 

## 🤔 Expected Behavior

All children of `Select` render.

## 😯 Current Behavior

Nothing renders.

## 🔦 Context

I am trying to style the `Button` inside the `Select` using the values of the `children` callback.

## 💻 Code Sample

https://codesandbox.io/p/sandbox/white-surf-4cv3bj

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-aria-components   | 1.0.0-alpha.3 |

**Comment by devongovett** [comment]
I think this is a duplicate of #4369, which was fixed by #4385. Could you try the nightly version?

**Comment by meesvandongen** [comment]
I can confirm it is working in `1.0.0-nightly.3906`
